packages/ui/src/ui/pages/navigation/tabs/Queue/views/Exports/ExportsEditDialog/ExportsEditDialog.tsx

```diff
@@ -0,0 +1,66 @@
+import React, {useEffect, useState} from 'react';
+import {useSelector} from 'react-redux';
+
+import {useFetchBatchQuery} from '../../../../../../../store/api/yt';
+import {
+    makeGetExportsParams,
+    useExportMutation,
+} from '../../../../../../../store/api/navigation/tabs/queue/queue';
+import {getPath} from '../../../../../../../store/selectors/navigation';
+
+import {FormApi, YTDFDialog} from '../../../../../../../components/Dialog';
+
+import {QueueExport, QueueExportConfig} from '../../../../../../../types/navigation/queue/queue';
+import {ExportConfigUtility} from '../Exports';
+
+import {fields, prepareInitialValues, prepareUpdateValues, validate} from './utils';
+
+interface DialogProps {
+    type: 'edit' | 'create';
+    visible: boolean;
+    prevConfig?: QueueExportConfig<number> & {export_name: string};
+    onClose: () => void;
+}
+
+export type ExportsFormValues = ExportConfigUtility & QueueExportConfig<{value: number}>;
+
+export function ExportsEditDialog(props: DialogProps) {
+    const {type, prevConfig, visible, onClose} = props;
+    const path = useSelector(getPath);
+
+    const {data: configs} = useFetchBatchQuery<QueueExport<number>>(makeGetExportsParams(path));
+    const [update, {isLoading}] = useExportMutation();
+
+    const [initialValues, setInitialValues] = useState<ExportsFormValues>();
+
+    useEffect(() => {
+        const rawValues = configs ? configs[0].output : {};
+        setInitialValues(prepareInitialValues(rawValues, prevConfig?.export_name));
+    }, [prevConfig?.export_name, configs]);
+
+    const onAdd = async (form: FormApi<ExportsFormValues, Partial<ExportsFormValues>>) => {
+        const {values} = form.getState();
+        const preparedValues = prepareUpdateValues(values);
+        await update({
+            prevConfig,
+            newConfig: {...preparedValues, export_name: values.export_name},
+            type: 'edit',
+        });
+    };
+
+    return (
+        <YTDFDialog<ExportsFormValues>
+            onAdd={onAdd}
+            visible={visible}
+            onClose={onClose}
+            size="l"
+            headerProps={{title: type === 'create' ? 'Create export' : 'Edit export'}}
+            initialValues={initialValues}
+            fields={fields}
```

Comment:
There is no reason to keep fields in a separate file, the less hops we have the easier it is to read.
It is a good approach when reading of your code is linear without any hops and back-scrollings.